### PR TITLE
simplify group-version-to-pkg-path

### PR DESCRIPTION
**Pull request by caesarxuchao**
simplify group-version-to-pkg-path, depends on #14156.

@lavalamp, please only take a look at the second commit. Thanks.

### CODE REVIEW

**Comment by lavalamp** [comment]
The second commit LGTM

**Comment by caesarxuchao** [comment]
Thanks. I will rebase after #14156 is merged.

**Comment by lavalamp** [comment]
@caesarxuchao can you rebase? thanks :)

**Comment by caesarxuchao** [comment]
rebased.

File: hack/lib/util.sh [hunk 1]
```diff
@@ -261,14 +259,8 @@ kube::util::group-version-to-pkg-path() {
...
-    api)
-      echo "api/v1"
-      ;;
-    api/*)
-      echo "${group_version}"
-      ;;
-    api/*)
-      echo "${group_version}"
+    unversioned)
+      echo "api/unversioned"
```
Comment (lavalamp) [inline hack/lib/util.sh]: We should probably move this package into apis/unversioned. :)

**Comment by lavalamp** [comment]
LGTM (I'm taking the passing test to mean that the generation scripts all still work!)

**Comment by caesarxuchao** [comment]
I manually tested the scripts. And our verify-generated.* scripts will complain if the behavior of generation scripts changes.

**Pull request merged**